src/simd_128.h

```diff
@@ -252,6 +252,35 @@ add_two_bufs(uint32_t* src, uint32_t* dest, size_t len, uint32_t card = F4)
     }
 }
 
+inline void sub_two_bufs(
+    uint32_t* bufa,
+    uint32_t* bufb,
+    uint32_t* res,
+    size_t len,
+    uint32_t card = F4)
+{
+    __uint128_t* _bufa = static_cast<__uint128_t*>(static_cast<void*>(bufa));
```

Comment:
Why this double cast?

Reply:
Once you cast in void*, you can static_cast in whatever your want, there is no more safety (you tricked the compiler). At this point, a reinterpret_cast is more readable/explicit and as the same level of safety.

Reply:
it's done